kernel/src/hil/eic.rs

```diff
@@ -27,10 +27,10 @@ pub trait ExternalInterruptController {
     /// Enables external interrupt on the given 'line'
     /// In asychronous mode, all edge interrupts will be
     /// interpreted as level interrupts and the filter is disabled.
-    fn line_enable(&self, line: &Self::Line, interrupt_mode: InterruptMode);
+    fn line_enable(&self, line: Self::Line, interrupt_mode: InterruptMode);
```

Comment:
`Line` is an associated type which can be in huge because we don't have any `trait bounds` on it such as `Copy`.
However, in the actual implementation, it is rather small but based on the trait it doesn't seem correct to me.